File: Server_Root/Server/RoomServer/GameEvent/GameEvents/CharacterStatusChange.cpp
#include "stdafx.h"
#include "CharacterStatusChange.h"
#include "../../../FBSFiles/CharacterStatusChange_generated.h"

namespace GameEvent
{
	CharacterStatusChange::CharacterStatusChange()
	{
	}

	CharacterStatusChange::~CharacterStatusChange()
	{
	}

	FBS::GameEventType CharacterStatusChange::GetType()
	{
		return FBS::GameEventType::GameEventType_CharacterStatusChange;
	}

	const char* CharacterStatusChange::Serialize(int* pLength)
	{
		auto statusField = m_Builder.CreateString(m_strStatusField);
		auto reason = m_Builder.CreateString(m_strReason);

		FBS::GameEvent::CharacterStatusChangeBuilder data_builder(m_Builder);
		data_builder.add_EntityID(m_nEntityID);
		data_builder.add_StatusField(statusField);
		data_builder.add_Reason(reason);
		data_builder.add_Value(m_fValue);
		auto data = data_builder.Finish();

		m_Builder.Finish(data);

		*pLength = m_Builder.GetSize();

		return (char*)m_Builder.GetBufferPointer();
	}
}

File: Client_Root/Client/Assets/Scripts/GameEvent/GameEvents/CharacterStatusChange.cs
using FlatBuffers;

namespace GameEvent
{
    public class CharacterStatusChange : IGameEvent
    {
        public int      m_nEntityID;
        public string   m_strStatusField;
        public string   m_strReason;
        public float    m_fValue;

        public override FBS.GameEventType GetEventType()
        {
            return FBS.GameEventType.CharacterStatusChange;
        }

        public override bool Deserialize(byte[] bytes)
        {
            var buf = new ByteBuffer(bytes);

            var data = FBS.GameEvent.CharacterStatusChange.GetRootAsCharacterStatusChange(buf);

            m_nEntityID = data.EntityID;
            m_strStatusField = data.StatusField;
            m_strReason = data.Reason;
            m_fValue = data.Value;

            return true;
        }

        public override string ToString()
        {
            return string.Format("{0}, EntityID : {1}, StatusField : {2}, Reason : {3}, Value : {4}", base.ToString(), m_nEntityID, m_strStatusField, m_strReason, m_fValue);
        }
    }
}
